utensor_cgen/transformer/graph_viz.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding:utf8 -*-
+r"""Graph Visualization Transformer
+
+Transformers that outputs a image file presenting the ugraph before the transformation
+"""
+import re
+from collections import defaultdict
+from copy import deepcopy
+
+from utensor_cgen.ir import OperationInfo, uTensorGraph
+from utensor_cgen.utils import parse_tensor_name
+
+from .base import Transformer
+
+__all__ = ["GraphVizTransformer"]
+
+
+class GraphVizTransformer(Transformer):
+  METHOD_NAME = 'graphViz'
+  KWARGS_NAMESCOPE = '_utensor_graphViz'
+  
+  def transform(self, ugraph):
```

Comment:
Two issues here:
1. As I said before, I think graph visualization should not be a part of `ugraph`. The main purpose of `ugraph` is to represent the structure and information needed to generate the NN code. We should refactor the viz part as another object or function which is responsible only for visualization. The other reason for doing it is that, if you take any library doing drawing/visualization as example, you'll find that there will be lots of parameters related to visualization. If we need to better visualize our `ugraph`, you can imagine how much parameters we can add to `ugraph` which are *irrelevant* to code generation if `ugraph` itself have to take care of visualization.
2. I suppose this transformer are designed to generate pictures of different stages of the `ugraph`, but it seems like you always overwrite the old pic with new one since they all share the same output file name. You need a way to write output to different files.

Reply:
Issues solved with my patch.

Reply:
@dboyliao    thanks for the improvement.
It looks like the `viz_graph()` is now part of the `Transformer` class
Any idea how we are going to visualize the graphs at different stages of a transform? (transformers other than `GraphVizTransformer`, as they won't inherit `viz_graph()`.

Reply:
In order to keep things reasonable and maintainable, I think it's good to keep a transformer atomic.
As a result, if you want to visualize multiple stages within one transformer, it's a good sign that we should refactor the transformer into multiple transformers.
After that, you just insert `graph_viz` transformer after the transformer of the stage you want to visualize.

Reply:
@dboyliao 
Transformers should be atomic. But, the graph viz is a powerful debugging tool. We often used it as the pdb for uGraph.

Reply:
Right.
But for now I think a transformator can serve it pretty well.
If you want to make graph viz works like a debugger, it requires more refactoring.